### PR TITLE
Add shared VGs and LVs

### PR DESCRIPTION
**Pull request by jgrund**
When a VG is present in a device tree, we can tell if it can be used on
another host by inspecting the underlying PVs.

If all PVs are present on another host, then we can presume the VG can
be available on that host as well.

Add a fn that will look at the PVs and add VGs / LVs to other hosts when
they match.

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

### CODE REVIEW

File: device-aggregator/src/linux_plugin_transforms.rs [hunk 1]
```diff
@@ -541,6 +541,76 @@ pub fn get_shared_pools<'a, S: ::std::hash::BuildHasher>(
...
+            })
+            .flatten()
+            .collect();
+
+        acc.append(&mut others);
+
+        acc
+    });
+
+    for (from_host, to_host, vg_name) in shared_vgs.into_iter() {
```
Comment (ip1981) [inline device-aggregator/src/linux_plugin_transforms.rs]: into_iter() might be implicit here?

File: device-aggregator/src/linux_plugin_transforms.rs [hunk 2]
```diff
@@ -541,6 +541,76 @@ pub fn get_shared_pools<'a, S: ::std::hash::BuildHasher>(
     shared_pools
 }
 
+/// Given some aggregated data
+/// Figure out what VGs can be shared between hosts and add them to the otehr hosts.
```
Comment (ip1981) [inline device-aggregator/src/linux_plugin_transforms.rs]:
```suggestion
/// Figure out what VGs can be shared between hosts and add them to the other hosts.
```

File: device-aggregator/src/linux_plugin_transforms.rs (hunk 1 above)
Comment (jgrund) [inline device-aggregator/src/linux_plugin_transforms.rs]:
```suggestion
    for (from_host, to_host, vg_name) in shared_vgs {
```

File: device-aggregator/src/linux_plugin_transforms.rs [hunk 3]
```diff
@@ -541,6 +541,76 @@ pub fn get_shared_pools<'a, S: ::std::hash::BuildHasher>(
...
+    let shared_vgs = xs.iter().fold(vec![], |mut acc, (from_host, x)| {
+        let mut others: Vec<_> = xs
+            .iter()
+            .filter(|(k, _)| k != &from_host)
+            .filter_map(|(to_host, y)| {
+                let shared_vgs: Vec<_> = x
+                    .vgs
+                    .iter()
+                    .filter(|(_, vg)| vg.pvs_major_minor.iter().all(|mm| y.devs.contains_key(mm)))
+                    .map(|(vg_name, _)| (from_host.clone(), to_host.clone(), vg_name.clone()))
```
Comment (utopiabound) [inline device-aggregator/src/linux_plugin_transforms.rs]: Clippy really doesn't like this line.
https://travis-ci.org/github/whamcloud/device-scanner/jobs/667850768

Comment (jgrund) [reply device-aggregator/src/linux_plugin_transforms.rs]: I need to add in github actions so we can see clippy complain

File: device-aggregator/src/linux_plugin_transforms.rs [hunk 4]
```diff
@@ -541,6 +548,104 @@ pub fn get_shared_pools<'a, S: ::std::hash::BuildHasher>(
...
+
+        acc.append(&mut others);
+
+        acc
+    });
+
+    for (from_host, to_host, vg_name) in shared_vgs {
+        let from = xs.get(&from_host).unwrap();
+
+        let vg = from.vgs.get(&vg_name).cloned().unwrap();
```
Comment (johnsonw) [inline device-aggregator/src/linux_plugin_transforms.rs]: Should we use `expect` instead of `unwrap` here?

Comment (jgrund) [reply device-aggregator/src/linux_plugin_transforms.rs]: We could.

**Pull request merged**